src/map/mob.c

```diff
@@ -1341,9 +1341,20 @@ static int mob_ai_sub_hard_slavemob(struct mob_data *md, int64 tick)
 		) {
 			short x = bl->x, y = bl->y;
 			mob_stop_attack(md);
-			if(map->search_freecell(&md->bl, bl->m, &x, &y, MOB_SLAVEDISTANCE, MOB_SLAVEDISTANCE, 1)
-				&& unit->walktoxy(&md->bl, x, y, 0))
-				return 1;
+			if (map->search_freecell(&md->bl, bl->m, &x, &y, MOB_SLAVEDISTANCE, MOB_SLAVEDISTANCE, 1)
+			    && unit->walktoxy(&md->bl, x, y, 0)) {
+				struct mob_data *m_md = BL_CAST(BL_MOB, bl);
+				nullpo_retr(1, m_md);
+				switch (m_md->state.skillstate) {
+				case MSS_BERSERK:
+				case MSS_ANGRY:
+				case MSS_RUSH:
+				case MSS_FOLLOW:
+					break;
+				default:
+					return 1; // master not involved in battle, no need to pick target
+				}
```

Comment:
probably better extract this code into separate function?
and also add config option for switch to use old and new way for slaves?

Reply:
Can do so, although my sources told me its official behavior.

Reply:
Done, check newer commits.